cmd/podman/pod_ps.go

```diff
@@ -0,0 +1,445 @@
+	NumberOfContainers bool
+}
+
+type podPsTemplateParams struct {
+	ID                 string
+	Name               string
+	Labels             string
+	Cgroup             string
+	UsePodCgroup       bool
+	NamesOfContainers  string
+	IdsOfContainers    string
+	NumberOfContainers int
+}
+
+// podPsJSONParams is used as a base structure for the psParams
+// If template output is requested, podPsJSONParams will be converted to
+// podPsTemplateParams.
+// podPsJSONParams will be populated by data from libpod.Container,
+// the members of the struct are the sama data types as their sources.
+type podPsJSONParams struct {
+	ID                 string     `json:"id"`
+	Name               string     `json:"name"`
+	Labels             fields.Set `json:"labels"`
+	Cgroup             string     `json:"cgroup"`
+	UsePodCgroup       bool       `json:"podcgroup"`
+	IdsOfContainers    []string   `json:"ctrids"`
+	NamesOfContainers  []string   `json:"ctrnames"`
+	NumberOfContainers int        `json:"numctrs"`
+}
+
+// Type declaration and functions for sorting the PS output
+type podPsSorted []podPsJSONParams
+
+func (a podPsSorted) Len() int      { return len(a) }
+func (a podPsSorted) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+
+type podPsSortedId struct{ podPsSorted }
+
+func (a podPsSortedId) Less(i, j int) bool { return a.podPsSorted[i].ID < a.podPsSorted[j].ID }
+
+type podPsSortedNumber struct{ podPsSorted }
+
+func (a podPsSortedNumber) Less(i, j int) bool {
+	return a.podPsSorted[i].NumberOfContainers < a.podPsSorted[j].NumberOfContainers
+}
+
+type podPsSortedName struct{ podPsSorted }
+
+func (a podPsSortedName) Less(i, j int) bool { return a.podPsSorted[i].Name < a.podPsSorted[j].Name }
+
+var (
+	podPsFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "ctr-names",
+			Usage: "Display the container names",
+		},
+		cli.BoolFlag{
+			Name:  "ctr-ids",
+			Usage: "Display the container UUIDs. If no-trunc is not set they will be truncated",
+		},
+		cli.StringFlag{
+			Name:  "filter, f",
+			Usage: "Filter output based on conditions given",
+		},
+		cli.StringFlag{
+			Name:  "format",
+			Usage: "Pretty-print pods to JSON or using a Go template",
+		},
+		cli.BoolFlag{
+			Name:  "no-trunc",
+			Usage: "Display the extended information",
+		},
+		cli.BoolFlag{
+			Name:  "quiet, q",
+			Usage: "Print the numeric IDs of the pods only",
+		},
+		cli.BoolFlag{
+			Name:  "cgroup",
+			Usage: "Print the Cgroup information of the pod",
+		},
+		cli.BoolFlag{
+			Name:  "labels",
+			Usage: "Print the labels of the pod",
+		},
+		cli.StringFlag{
+			Name:  "sort",
+			Usage: "Sort output by id, name, or number",
+			Value: "name",
+		},
+	}
+	podPsDescription = "Prints out information about pods"
+	podPsCommand     = cli.Command{
+		Name:                   "ps",
+		Usage:                  "List pods",
+		Description:            podPsDescription,
+		Flags:                  podPsFlags,
+		Action:                 podPsCmd,
+		ArgsUsage:              "",
```

Comment:
If you don't take positional arguments (and this command doesn't seem to), you should be able to leave `ArgsUsage` unset (by dropping this line) instead of explicitly setting it to an empty string.